### PR TITLE
Rewrite LibreNMS Processor support

### PR DESCRIPTION
**Pull request by murrant**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

File: includes/common.php [hunk 1]
```diff
@@ -1888,3 +1888,11 @@ function array_by_column($array, $column)
 {
     return array_combine(array_column($array, $column), $array);
 }
+
+/**
+ * Dump and die
+ */
+function dd() {
```
Comment (laf) [inline includes/common.php]: Wrap this in a function exists check for when we switch to Laravel?

**Comment by laf** [comment]
How come you've not used the yaml added that's commented out?

File: includes/common.php (hunk 1 above)
Comment (murrant) [reply includes/common.php]: @laf sorry, this was for someone to test.  Not ready for review fully yet :)

**Comment by murrant** [comment]
That was for testing hrProcessors are discovered by default ;)

**Pull request closed without merging**